src/softmax/problem_description.cpp

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/softmax/problem_description.hpp>
+#include <miopen/names.hpp>
+
+#include <sstream>
+
+namespace miopen {
+
+namespace softmax {
+
+NetworkConfig ProblemDescription::MakeNetworkConfig() const
+{
+    std::string network_config = "sfmfwd-";
+
+    if(isForward)
+    {
+        int n_x, c_x, h_x, w_x;
+        int n_y, c_y, h_y, w_y;
+
+        std::tie(n_x, c_x, h_x, w_x) = tien<4>(xdxDesc.GetLengths());
+        std::tie(n_y, c_y, h_y, w_y) = tien<4>(yDesc.GetLengths());
+
+        network_config += "n_x" + std::to_string(n_x) + "c_x" + std::to_string(c_x) + "h_x" +
```

Comment:
Rewrite this stuff using `ostringstream`, see https://github.com/ROCm/MIOpen/blob/c7fa6f7fbec5f2a76128e2c9ae80f66c802722a6/src/conv/problem_description.cpp#L106

Reply:
@atamazov 
Merging strings in this style were in initial code which is in develop currently. Scope of this ticket was to introduce solver infrastructure for softmax. Could we improve this in additional ticket / PR ?

Reply:
@Vsevolod1983 The "problem description" class is introduced in this PR, so I would prefer fixing right here. However, I won't insist if you would like to postpone this for a while.

**_This PR can be merged as is provided that the next your PR is the one that resolves this review comment._**

Reply:
Agreed to make the fix in the next PR.

Reply:
Discussed with @atamazov . PR can be merged.